client/app/collections/filtered.coffee

```diff
@@ -64,6 +64,16 @@ module.exports = class FilteredCollection
         if query
             res = app.contacts.filter filter @query
 
+            # Reduce to get max score and get a median limit, then excludes all
+            # results $lt it.
+            maxScore = res.reduce (currentMax, model) ->
+                Math.max currentMax, scores.get model
+            , 0
+
+            if maxScore > 5
+                median = maxScore / 2
```

Comment:
You know that the median is not the maximum / 2. Maybe calling it middle would be more clear ;-)